modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CClientCodegen.java

```diff
@@ -0,0 +1,700 @@
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openapitools.codegen.languages;
+
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.Schema;
+import org.apache.commons.lang3.StringUtils;
+import org.openapitools.codegen.*;
+import org.openapitools.codegen.utils.ModelUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Map;
+
+public class CClientCodegen extends DefaultCodegen implements CodegenConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CClientCodegen.class);
+    public static final String GEM_NAME = "gemName";
+    public static final String MODULE_NAME = "moduleName";
+    public static final String GEM_VERSION = "gemVersion";
+    public static final String GEM_LICENSE = "gemLicense";
+    public static final String GEM_REQUIRED_RUBY_VERSION = "gemRequiredRubyVersion";
+    public static final String GEM_HOMEPAGE = "gemHomepage";
+    public static final String GEM_SUMMARY = "gemSummary";
+    public static final String GEM_DESCRIPTION = "gemDescription";
+    public static final String GEM_AUTHOR = "gemAuthor";
+    public static final String GEM_AUTHOR_EMAIL = "gemAuthorEmail";
+
+    protected String gemName;
+    protected String moduleName;
+    protected String gemVersion = "1.0.0";
+    protected String specFolder = "spec";
+    protected String libFolder = "lib";
+    protected String gemLicense = "proprietary";
+    protected String gemRequiredRubyVersion = ">= 1.9";
+    protected String gemHomepage = "http://org.openapitools";
+    protected String gemSummary = "A ruby wrapper for the REST APIs";
+    protected String gemDescription = "This gem maps to a REST API";
+    protected String gemAuthor = "";
+    protected String gemAuthorEmail = "";
+    protected String apiDocPath = "docs/";
+    protected String modelDocPath = "docs/";
+
+    protected static int emptyMethodNameCounter = 0;
+
+    public CClientCodegen() {
+        super();
+
+        // clear import mapping (from default generator) as ruby does not use it
+        // at the moment
+        importMapping.clear();
+
+        modelPackage = "models";
+        apiPackage = "api";
+        outputFolder = "generated-code" + File.separator + "C";
```

Comment:
maybe generated-code/C-LibCurl

Reply:
Sure

Reply:
Thanks for the feedback. I've switched it to `generated-code/C-libcurl`.